components/proxy/src/main/java/com/hotels/styx/Environment.java

```diff
@@ -33,12 +35,14 @@
 public final class Environment implements com.hotels.styx.api.Environment {
     private final Version version;
     private final EventBus eventBus;
+    private final EventNexus eventNexus;
```

Comment:
It's definitely *not* necessary to have both. In fact the `EventNexus` can replace a lot of our current code.
In this PR I only used it to add new functionality in order to avoid making the PR too large.

There are three alternatives:
* Replace `EventBus` with `EventNexus` in a separate PR, also create PRs for other places it can be used
* Add more changes to this PR, either just to replace `EventBus` or every possible change
* Cancel this PR, create a separate PR to introduce `EventNexus` and change current code to use it, then do the proxy-start-endpoint feature as a new PR after that one is merged.

What do you think is the best approach?

Reply:
Our specific requirement is to implement an admin interface endpoint for probing the proxy server status. That is the first priority.

You don't need to cancel this PR. Just back off the event nexus changes, and create a separate branch for them.

Regardless, it was good to see that code in this PR as it gives us some thoughts and direction to work on.

Reply:
The Event Nexus provides the implementation, if it is backed off, the functionality will not work. Are you proposing an alternative implementation?

Reply:
I image you have implemented EventNexus since it was necessary for this functionality. We might have to discuss that case... Otherwise I agree with mikko that replacing EventBus with the new class could be a separate PR.

In any case, as discussed below, it doesn't seem that that feeding the admin endpoint directly with the events is the best solution here.  At the very least, we'd need to make sure we  are not going to lose any events, but it seems that a 'centralised registry' could be a better option.

Reply:
Yeah, for this use case, a "centralised registry" is perfectly fine.

However for the general use case the consumers of the "centralised registry" may want to get notified of the state changes. So that is where & when I would bring back the event nexus.